apps/omg_watcher/lib/omg_watcher/block_getter.ex

```diff
@@ -136,10 +136,8 @@ defmodule OMG.Watcher.BlockGetter do
 
     case Core.validate_executions(tx_exec_results, block_application, state) do
       {:ok, state} ->
-        :ok =
-          {:child_chain, "block.get"}
-          |> OMG.Bus.Event.new(:block_received, block_application)
-          |> OMG.Bus.direct_local_broadcast()
+        if Code.ensure_loaded?(OMG.WatcherInfo.DB.Block),
+          do: Kernel.apply(OMG.WatcherInfo.BlockApplicator, :insert_block!, [block_application])
```

Comment:
I need you to rethink this.

Above this line theres:
```
tx_exec_results = for(tx <- block_application.transactions, do: OMG.State.exec(tx, :ignore_fees))
```

Which applies the block to the Watcher ledger. After this, you're sending the block to postgres.
What happens if PG can't swallow this block?


In `OMG.Watcher.Depositor` you're first applying the deposits to PG and then on RocksDB. You don't have a consistent approach in solving this ticket. You need to consider failure! How are you going to keep two databases in sync?

Reply:
Here the approach is similar to the mentioned deposit. 
The code you mentioned does the stateful validation of the block's transactions but does not persist the legder and block's data into RocksDB yet. It is done in the [next step](https://github.com/omgnetwork/elixir-omg/blob/972c70db1b9b1dcfd521d51dfcbde7631eb0a2ff/apps/omg_watcher/lib/omg_watcher/block_getter.ex#L170)

So let consider the scenarios here:

1) __PG write fails__
The in-memory `State` already contains new block transactions applied but RocksDB is one block behind. ([RocksDB update](https://github.com/omgnetwork/elixir-omg/blob/972c70db1b9b1dcfd521d51dfcbde7631eb0a2ff/apps/omg_watcher/lib/omg_watcher/block_getter.ex#L170) is in the following `handle_continue`)

Failure of `WatcherInfo` call will result of  `BlockGetter` process death which also brings down the `State` process (see: [OMG.Watcher.BlockGetter.Supervisor](https://github.com/omgnetwork/elixir-omg/blob/972c70db1b9b1dcfd521d51dfcbde7631eb0a2ff/apps/omg_watcher/lib/omg_watcher/block_getter/supervisor.ex)).
Both processes will be restarted by the :point_up: supervisor - so `State` gets RocksDB version of the ledger.


2) __PG write succeeds and RocksDB update fails__
`BlockGetter` and `State` are being restarted and gets the state from the previous block. PG already has the current block's data. Processing the block again WatcherInfo code gets executed again but fortunately [its idempotent]() - so PG will stay unchanged.

Reply:
Okay, it makes sense!
```
Processing the block again WatcherInfo code gets executed again but fortunately its idempotent - so PG will stay unchanged.
```

are you sure about this tho?

Reply:
Fortunately it's tested already (I added the test for block insert indempothency)
- [Exits & deposits writes](https://github.com/omgnetwork/elixir-omg/blob/972c70db1b9b1dcfd521d51dfcbde7631eb0a2ff/apps/omg_watcher_info/test/omg_watcher_info/db/eth_event_test.exs#L276)
- [Block applicator test](https://github.com/omgnetwork/elixir-omg/blob/972c70db1b9b1dcfd521d51dfcbde7631eb0a2ff/apps/omg_watcher_info/test/omg_watcher_info/block_applicator_test.exs#L54)